addons/crm/wizard/crm_lead_to_opportunity.py

```diff
@@ -12,6 +12,11 @@ class Lead2OpportunityPartner(models.TransientModel):
 
     @api.model
     def default_get(self, fields):
+        # JEM TDE FIXME: clean that brol
+        """ Check some preconditions before the wizard executes. """
+        for lead in self.env['crm.lead'].browse(self._context.get('active_ids', [])):
+            if lead.probability == 100:
+                raise UserError(_("Closed/Dead leads cannot be converted into opportunities."))
```

Comment:
does that even work now that the web client doesn't call `default_get` anymore? it's supposed to trigger *before* the form opens...

Reply:
While moving that brol, we should probably have a constraint checking lead_id probability instead of overriding default_get .

Reply:
@odony 
Yes, it is working, the `onchange` still call the default_get with the correct context.

@tde-banana-odoo 
A constraint is only check when we save the record, which it is too late in this situation (we want to trigger when we open the wizard, not after fill it).

Moreover, this corner case (when lead has a probability of 100 %) is not correctly cover (but it is not the scope of these changes)
![image](https://user-images.githubusercontent.com/53175387/147343824-0771cb2c-99bc-4d9d-a67d-6807e3558a6d.png)
(2 lost buttons :thinking: and when I click on "Convert to Opportunity" I get the error)

Then I will keep it in the same place with a cleaner version of it.